primitives/src/lib.rs

```diff
@@ -43,6 +43,9 @@ pub type TextFormat = Vec<u8>;
 /// The type for looking up accounts. We don't expect more than 4 billion of them.
 pub type AccountIndex = u32;
 
+/// The type of marketplace Id
+pub type MarketplaceId = u32;
```

Comment:
put it inside a mod.
pub mod marketplace {
    pub type MarketplaceId = u32;
}

This way it's going to be much easier to know what type is from which pallet.